core/src/main/scala/org/locationtech/rasterframes/expressions/localops/Resample.scala

```diff
@@ -22,53 +22,168 @@
-  override protected def op(left: Tile, right: Double): Tile = left.resample((left.cols * right).toInt,
-                                                                             (left.rows * right).toInt, NearestNeighbor)
-  override protected def op(left: Tile, right: Int): Tile = op(left, right.toDouble)
+  override def dataType: DataType = left.dataType
+  override def children: Seq[Expression] = Seq(left, right, method)
+
+  def targetFloatIfNeeded(t: Tile, method: ResampleMethod): Tile =
+     method match {
+      case NearestNeighbor | Mode | RMax | RMin | Sum  ⇒ t
+      case _ ⇒ fpTile(t)
+     }
+
+  def stringToMethod(methodName: String): ResampleMethod =
+    methodName.toLowerCase().trim().replaceAll("_", "") match {
+      case "nearestneighbor" | "nearest" ⇒ NearestNeighbor
+      case "bilinear" ⇒ Bilinear
+      case "cubicconvolution" ⇒ CubicConvolution
+      case "cubicspline" ⇒ CubicSpline
+      case "lanczos" | "lanzos" ⇒ Lanczos
+      // aggregates
+      case "average" ⇒ Average
+      case "mode" ⇒ Mode
+      case "median" ⇒ Median
+      case "max" ⇒ RMax
+      case "min" ⇒ RMin
+      case "sum" ⇒ Sum
+    }
+
+   // These methods define the core algorithms to be used.
+  def op(left: Tile, right: Tile, method: String): Tile = {
+    val m = stringToMethod(method)
+    targetFloatIfNeeded(left, m)
+      .resample(right.cols, right.rows, m)
+  }
+
+  def op(left: Tile, right: Double, method: String): Tile = {
+    val m = stringToMethod(method)
+    targetFloatIfNeeded(left, m)
+      .resample((left.cols * right).toInt, (left.rows * right).toInt, m)
+  }
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    // copypasta from BinaryLocalRasterOp
+    if (!tileExtractor.isDefinedAt(left.dataType)) {
+      TypeCheckFailure(s"Input type '${left.dataType}' does not conform to a raster type.")
+    }
+    else if (!tileOrNumberExtractor.isDefinedAt(right.dataType)) {
+      TypeCheckFailure(s"Input type '${right.dataType}' does not conform to a compatible type.")
+    } else method.dataType match {
+      case StringType ⇒ TypeCheckSuccess
+      case _ ⇒ TypeCheckFailure(s"Cannot interpret value of type `${method.dataType.simpleString}` for resampling method; please provide a String method name.")
+    }
+  }
+
+  override def nullSafeEval(input1: Any, input2: Any, input3: Any): Any = {
+    // more copypasta from BinaryLocalRasterOp
+    implicit val tileSer = TileUDT.tileSerializer
+
+    val (leftTile, leftCtx) = tileExtractor(left.dataType)(row(input1))
+    val methodString = input3.asInstanceOf[UTF8String].toString
+
+    val result: Tile = tileOrNumberExtractor(right.dataType)(input2) match {
+      // in this case we expect the left and right contexts to vary. no warnings raised.
+      case TileArg(rightTile, _) ⇒ op(leftTile, rightTile, methodString)
+      case DoubleArg(d) ⇒ op(leftTile, d, methodString)
+      case IntegerArg(i) ⇒ op(leftTile, i.toDouble, methodString)
+    }
```

Comment:
I think I would argue for doing the `stringToMethod` in here and having the `op` methods take the real type. You've "reified" the other inputs already, seems like we should do it here. The issue we were discussing before relating to keeping it "stringly typed" until the last moment was a) partially due to me misreading the exception, and b) only applies to values making their way into the member fields of the class (where they'd get serialized).